src/run/run_step.go

```diff
@@ -97,6 +97,10 @@ func run(ctx context.Context, state *core.BuildState, label core.BuildLabel, arg
 			log.Fatalf("Can't find binary %s", splitCmd[0])
 		}
 		splitCmd[0] = cmd
+		if debug {
+			// Obviously this only works if we really are a java -jar
```

Comment:
plz should not have hardcoded knowledge of Java flag incantations in it... can this not be included on the target in its debug config?

Reply:
That would be ideal but I couldn't see how to specify that. We're creating this command using `core.ReplaceSequences(state, target, fmt.Sprintf("$(out_exe %s)", target.Label))` which sticks a `java -jar` on the front of [output], so this layer violation is already happening. The commands in `java_binary` seem to be only used for generating the jar (they seem to be jarcat commands)?